jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/CreateQueueDialog.kt

```diff
@@ -43,24 +43,26 @@ class CreateQueueDialog(
     // TODO: Override cancel action when telemetry added
 
     override fun doOKAction() {
-        if (isOKActionEnabled) {
-            setOKButtonText(message("sqs.create.queue.in_progress"))
-            isOKActionEnabled = false
-
-            launch {
-                try {
-                    createQueue()
-                    runInEdt(ModalityState.any()) {
-                        close(OK_EXIT_CODE)
-                    }
-                    project.refreshAwsTree(SqsResources.LIST_QUEUE_URLS)
-                } catch (e: Exception) {
-                    // API only throws QueueNameExistsException if the request includes attributes whose values differ from those of the existing queue.
-                    LOG.warn(e) { message("sqs.create.queue.failed", queueName()) }
-                    setErrorText(e.message)
-                    setOKButtonText(message("sqs.create.queue.create"))
-                    isOKActionEnabled = true
+        if (!isOKActionEnabled) {
+            return
+        }
+
+        setOKButtonText(message("sqs.create.queue.in_progress"))
+        isOKActionEnabled = false
+
+        launch {
+            try {
+                createQueue()
+                runInEdt(ModalityState.any()) {
```

Comment:
mix of coroutines and runInEdt seems odd

Reply:
Yeah that is weird, changed